### PR TITLE
Lazy create RequestServicesFeature DisposeAsync statemachine

### PR DESCRIPTION
**Pull request by benaadams**
image

Bunch of work for it currently 

image

### CODE REVIEW

**Comment by benaadams** [comment]
Code check hates me

File: src/Http/Http/src/Features/RequestServicesFeature.cs [hunk 1]
```diff
@@ -42,12 +42,16 @@ public IServiceProvider RequestServices
...
         {
             switch (_scope)
             {
                 case IAsyncDisposable asyncDisposable:
-                    await asyncDisposable.DisposeAsync();
+                    var vt = asyncDisposable.DisposeAsync();
+                    if (!vt.IsCompletedSuccessfully)
+                    {
+                        return Awaited(this, vt);
+                    }
```
Comment (benaadams) [inline src/Http/Http/src/Features/RequestServicesFeature.cs]: Should trigger the result here

**Comment by Tratcher** [comment]
@benaadams why do you keep opening and closing your PRs? To re-trigger CI due to flaky tests? If so, let us handle that, we want to track those and get them fixed. Also, we can re-run individual builds and get things though quicker.

**Comment by benaadams** [comment]
>  To re-trigger CI due to flaky tests?

Flaky 3rd party infrastructure; failed downloads, invalid responses or cert validation way before the tests. Guessing its just clashing with an update.

**Comment by rynowak** [comment]
> 
> 
> Code check hates me

Code check hates everyone

**Review by Tratcher — APPROVED** [review]

**Pull request merged**